zerver/lib/user_groups.py

```diff
@@ -9,12 +9,10 @@
 from zerver.models import Realm, UserGroup, UserGroupMembership, UserProfile
 
 
-def access_user_group_by_id(
-    user_group_id: int, user_profile: UserProfile, for_mention: bool = False
-) -> UserGroup:
+def access_user_group_by_id(user_group_id: int, user_profile: UserProfile) -> UserGroup:
     try:
         user_group = UserGroup.objects.get(id=user_group_id, realm=user_profile.realm)
-        if not for_mention and user_group.is_system_group:
+        if user_group.is_system_group:
             raise JsonableError(_("Insufficient permission"))
```

Comment:
@sahil839 for thoughts, but I think we still want the `for_mention` flag in that as we modify the user groups system to be used more widely for permissions, I expect us to make a "who can mention this group" setting, and so it's helpful to preserve the distinction in the interface.

Reply:
We currently do not use `access_user_group_by_id` in the mention code, but I think it can be used when we add the "who can mention this group" setting.

Reply:
My goal was to make a quick change for correctness and not leave a bad implication by the `for_mention` flag existing, but _not being used for mentions_.  I don't have time to switch the mention code to use this codepath, so this seemed like the fastest and clearest fix.  If someone wants to take this the other direction, and fix `access_user_group_by_id` to return true if `for_mention` after loading `user_group`, and also make it be used in the mention codepath, I'm all for that.  But I don't want it to leave the `for_mention` flag named that and not used for mentions.

Reply:
We currently use name of the user group in mention codepath, not ID, to get the user group object and also since there is no current restriction on mentioning user groups, I think there is no reason to use `access_user_group_by_id` in mention codepath for now.
We may use ID of user group in future, for eg - if we allowed user groups with same name in future, but I am not sure.

Reply:
Given the context that we don't use this in the mention code path and may not do so in the future, I think it makes sense to merge this as written.